stake-pool/cli/src/main.rs

```diff
@@ -660,8 +688,125 @@ fn command_deposit(
             &validator_stake_account,
             &stake_pool.reserve_stake,
             &token_receiver,
+            &stake_pool.manager_fee_account,
+            &referrer,
+            &stake_pool.pool_mint,
+            &spl_token::id(),
+        )
+    };
+
+    instructions.append(&mut deposit_instructions);
+
+    let mut transaction =
+        Transaction::new_with_payer(&instructions, Some(&config.fee_payer.pubkey()));
+
+    let (recent_blockhash, fee_calculator) = config.rpc_client.get_recent_blockhash()?;
+    check_fee_payer_balance(
+        config,
+        total_rent_free_balances + fee_calculator.calculate_fee(transaction.message()),
+    )?;
+    unique_signers!(signers);
+    transaction.sign(&signers, recent_blockhash);
+    send_transaction(config, transaction)?;
+    Ok(())
+}
+
+fn command_deposit_sol(
+    config: &Config,
+    stake_pool_address: &Pubkey,
+    from: &Option<Keypair>,
+    token_receiver: &Option<Pubkey>,
+    referrer: &Option<Pubkey>,
+    amount: f64,
+) -> CommandResult {
+    if !config.no_update {
+        command_update(config, stake_pool_address, false, false)?;
+    }
+
+    let amount = spl_token::ui_amount_to_amount(amount, 9);
+
+    // Check withdraw_from balance
+    let from_pubkey = from.as_ref().map_or_else(
+        || config.fee_payer.try_pubkey().unwrap(),
+        |keypair| keypair.try_pubkey().unwrap(),
+    );
+    let from_balance = config.rpc_client.get_balance(&from_pubkey)?;
+    if from_balance < amount {
+        return Err(format!(
+            "Not enough SOL to deposit into pool: {}.\nMaximum deposit amount is {} SOL.",
+            spl_token::amount_to_ui_amount(amount, 9),
+            spl_token::amount_to_ui_amount(from_balance, 9)
+        )
+        .into());
+    }
+
+    let stake_pool = get_stake_pool(&config.rpc_client, stake_pool_address)?;
+
+    let mut instructions: Vec<Instruction> = vec![];
```

Comment:
For added safety, let's craft this transaction as:

* create ephemeral keypair
* transfer `amount` from `from` to the ephemeral keypair
* pass the ephemeral keypair to `sol_deposit` and sign with it

this way, people can feel better and know that the maximum amount at risk is however much is transferred to the ephemeral keypair, and *not* everything in their account